hail/src/test/scala/is/hail/io/compress/BGzipCodecSuite.scala

```diff
@@ -114,4 +115,57 @@ class BGzipCodecSuite extends SparkSuite {
     }
     p.check()
   }
+
+  @Test def testVirtualSeek() {
+    sc.hadoopConfiguration.setLong("mapreduce.input.fileinputformat.split.minsize", 1L)
+    // real offsets of the start of each block
+    val blockStarts = Array[Long](14653, 28034, 40231, 55611, 69140, 82949, 96438, 110192, 121461, 133703,
+      146664, 158711, 171239, 181362)
+    // offsets into the uncompressed file
+    val uncompBlockStarts = Array[Long](65280, 130560, 195840, 261120, 326400, 391680, 456960, 522240,
+      587520, 652800, 718080, 783360, 848640, 913920)
+
+    val blockPointerOffset = 16 // magic number for creating virtual file pointers into bgzipped files
+
+    val uncompPath = "src/test/resources/sample.vcf"
+    val compPath = "src/test/resources/sample.vcf.gz"
+
+    val uncompHPath = new hd.fs.Path(uncompPath)
+    val compHPath = new hd.fs.Path(compPath)
+
+    val fs = uncompHPath.getFileSystem(hadoopConf)
+    val compIS = fs.open(compHPath)
+    val uncompIS = fs.open(uncompHPath)
+
+    val decompIS = new BGzipInputStream(compIS)
+
+    for ((cOff, uOff) <- blockStarts.zip(uncompBlockStarts);
+         extra <- Seq(0, 50)) {
```

Comment:
These cases seem somewhat redundant.  I'd ask to do: the first block, an interior block, the last block, the first byte, and interior byte, and the last byte.

Also, if you want to be compete, check seeing to the end of file and verify you're at end of file (read and get -1).

Reply:
We have no way to check for EOF other than reading and getting -1. I have currently specifically chosen to disallow seeking to one byte past the end of a block (the first byte of the next block). This prevents seeking directly to the end of the file. I'll fixup the cases to have better coverage though.

Reply:
`BlockCompressedFilePointerUtils.makeFilePointer(lastBlockOffset, lastBlockUncompressedLength)`

Reply:
For every other block, the set of positions are [(blockStart, 0), (blockStart, blockLength)) -- that is, non-inclusive of the block length.  This should be the same for the last block, in which case the (virtual) offset of end-of-file should be (compFileSize, 0).